include/tudat/astro/system_models/geometryUtils.h

```diff
@@ -0,0 +1,201 @@
+*/
+#ifndef TUDAT_GEOMETRYUTILS_H
+#define TUDAT_GEOMETRYUTILS_H
+
+#include <map>
+#include <iostream>
+#include <memory>
+
+namespace tudat
+{
+
+namespace system_models
+{
+class Triangle2d
+{
+private:
+    Eigen::Vector2d vertexA_, vertexB_, vertexC_;
+
+public:
+   // default constructor
+   Triangle2d( ):
+   vertexA_( Eigen::Vector2d::Zero() ),
+   vertexB_( Eigen::Vector2d::Zero() ),
+   vertexC_( Eigen::Vector2d::Zero() )
+   { }
+
+   Triangle2d( const Eigen::Vector2d& vertexA,
+               const Eigen::Vector2d& vertexB,
+               const Eigen::Vector2d& vertexC ): 
+               vertexA_( vertexA ), 
+               vertexB_( vertexB ), 
+               vertexC_( vertexC ) 
+   { }
+
+   Eigen::Vector2d getVertexA( ) const 
+   {
+       return vertexA_;
+   }
+   Eigen::Vector2d getVertexB( ) const
+   {
+       return vertexB_;
+   }
+   Eigen::Vector2d getVertexC( ) const
+   {
+       return vertexC_;
+   }
+};
+
+class Triangle3d
+{
+private:
+    Eigen::Vector3d vertexA_, vertexB_, vertexC_;
+    Eigen::Vector3d versorL_, versorM_;
+
+public:
+   Triangle3d( ) :
+   vertexA_( Eigen::Vector3d::Zero( ) ),
+   vertexB_( Eigen::Vector3d::Zero( ) ),
+   vertexC_( Eigen::Vector3d::Zero( ) ),
+   versorL_( Eigen::Vector3d::Zero( ) ),
+   versorM_( Eigen::Vector3d::Zero( ) )
+   { }
+   
+   Triangle3d( const Eigen::Vector3d& vertexA,
+               const Eigen::Vector3d& vertexB,
+               const Eigen::Vector3d& vertexC ): 
+               vertexA_( vertexA ),
+               vertexB_( vertexB ),
+               vertexC_( vertexC ) 
+   {  
+        // compute unit vectors l, m
+        versorL_ = ( vertexB_ - vertexA_ ).normalized( ) ;
+        Eigen::Vector3d edgeAB = vertexB_ - vertexA_;
+        Eigen::Vector3d edgeAC = vertexC_ - vertexA_;
+        Eigen::Vector3d normal = edgeAB.cross(edgeAC).normalized( );
+        versorM_ = normal.cross( versorL_ ).normalized( ); 
+    }
+
+   Eigen::Vector3d getVertexA( ) const
+   {
+       return vertexA_;
+   }
+   Eigen::Vector3d getVertexB( ) const
+   {
+       return vertexB_;
+   }
+   Eigen::Vector3d getVertexC( ) const
+   {
+       return vertexC_;
+   }
+   Eigen::Vector3d getVersorL( ) const
+   {
+       return versorL_;
+   }
+   Eigen::Vector3d getVersorM( ) const
+   {
+       return versorM_;
+   }
+};
+
+class ParallelProjection
+{
+private:
+    double minCoordinateL_, maxCoordinateL_, minCoordinateM_, maxCoordinateM_;
+    Triangle2d triangle2d_;
+    std::vector< double > lambdas_;
+    
+public:
+    ParallelProjection( ):  minCoordinateL_( 0.0 ), maxCoordinateL_( 0.0 ), minCoordinateM_( 0.0 ), maxCoordinateM_( 0.0 ),
+                            triangle2d_( ), lambdas_( )
+    { }
+
+    ParallelProjection( const Triangle3d& triangle, const Triangle3d& triangleToBeProjected, 
+                        const Eigen::Vector3d& projectionDirection_= Eigen::Vector3d::Zero( ) )
+    {
+        Eigen::Vector3d dummyDirection = Eigen::Vector3d::Zero( );

```

Comment:
why define as a separate variable (and not instantiate directly where it is used?)